core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -54,16 +54,20 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
     // Validate the configurations.
     val configNamesToExclude = excludedConfigs(topic, topicConfig)
 
-    val logs = logManager.logsByTopic(topic).toBuffer
-    if (logs.nonEmpty) {
-      /* combine the default properties with the overrides in zk to create the new LogConfig */
-      val props = new Properties()
-      topicConfig.asScala.foreach { case (key, value) =>
-        if (!configNamesToExclude.contains(key)) props.put(key, value)
+    def updateLogConfig: Unit = {
```

Comment:
Inner defs are nice when they are small. Here I wonder if we should just turn this into a `private def` instead so we can avoid cluttering the logic. Same for `updateThrottledList`.

Similar comment in `DynamicBrokerConfig`.

Reply:
Done.